build_projects/shared-build-targets-utils/Utils/HostVersion.cs

```diff
@@ -10,39 +10,61 @@ public class HostVersion : Version
         // are baked into the binary and is used to look up a serviced binary replacement.
         //
 
+        public struct VerInfo
+        {
+            public int Major;
+            public int Minor;
+            public int Patch;
+            public string Release;
+            public string BuildMajor;
+            public string BuildMinor;
+
+            public VerInfo(int major, int minor, int patch, string release, string buildMajor, string buildMinor)
+            {
+                Major = major;
+                Minor = minor;
+                Patch = patch;
+                Release = release;
+                BuildMajor = buildMajor;
+                BuildMinor = buildMinor;
+            }
+
+            public string WithoutSuffix => $"{Major}.{Minor}.{Patch}";
+
+            public string ToString()
+            {
+                return $"{Major}.{Minor}.{Patch}-{Release}-{BuildMajor}-{BuildMinor}";
+            }
+        }
         //
         // Latest hosts for production of nupkgs.
         //
 
-        // Version constants without suffix
-        public override int Major => 1;
-        public override int Minor => 0;
-        public override int Patch => 1;
-        public override string ReleaseSuffix => "rc3";
-        public string LatestHostVersionNoSuffix => $"{Major}.{Minor}.{Patch}";
-        public string LatestHostFxrVersionNoSuffix => $"{Major}.{Minor}.{Patch}";
-        public string LatestHostPolicyVersionNoSuffix => $"{Major}.{Minor}.{Patch}";
-        public string LatestHostPrerelease => ReleaseSuffix;
-        public string LatestHostBuildMajor => $"{CommitCountString}";
-        public string LatestHostBuildMinor => "00";
-        public string LatestHostSuffix => $"{ReleaseSuffix}-{LatestHostBuildMajor}-{LatestHostBuildMinor}";
-
         // Full versions and package information.
-        public string LatestHostVersion => $"{LatestHostVersionNoSuffix}-{LatestHostSuffix}";
-        public string LatestHostFxrVersion => $"{LatestHostFxrVersionNoSuffix}-{LatestHostSuffix}";
-        public string LatestHostPolicyVersion => $"{LatestHostPolicyVersionNoSuffix}-{LatestHostSuffix}";
-        public Dictionary<string, string> LatestHostPackages => new Dictionary<string, string>()
+        public string LatestHostPrerelease => "rc3";
```

Comment:
Other repos are expecting to get built in the pipeline and the pipeline infra enforces stable versions. Also RC4 is for the dev branch while RC3 is for release branch.

@ericstj should chime in here as well since he advised on a similar issue earlier this week. Effectively, stable versions get enforced automatically in the build via the infrastructure.